luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/AbridgedLogicalTableGroupFactory.java

```diff
@@ -0,0 +1,97 @@
+import com.yahoo.bard.webservice.data.metric.MetricDictionary;
+import com.yahoo.bard.webservice.data.time.DefaultTimeGrain;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
+import com.yahoo.bard.webservice.table.LogicalTable;
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.TableGroup;
+import com.yahoo.bard.webservice.table.TableIdentifier;
+import org.joda.time.DateTimeZone;
+
+import java.util.LinkedHashSet;
+import java.util.Locale;
+
+/**
+ * A factory that is used by default to support LogicalTableGroup.
+ */
+public class AbridgedLogicalTableGroupFactory implements Factory<LogicalTableGroup> {
+    private static final String ENTITY_TYPE = "abridged logical table group";
+
+    /**
+     * Build a LogicalTable instance.
+     *
+     * @param name  the config dictionary name (normally the apiName)
+     * @param configTable  ObjectNode that points to the value of corresponding table entry in config file
+     * @param resourceFactories  the source for locating dependent objects
+     *
+     * @return  A newly constructed config instance for the name and config provided
+     */
+    @Override
+    public LogicalTableGroup build(String name, ObjectNode configTable, LuthierIndustrialPark resourceFactories) {
+        validateFields(name, configTable);
+
+        LogicalTableGroup tableDictionary = new LogicalTableGroup();
+        LogicalTableName tableName = LogicalTableName.forName(name);
+        MetricDictionary metricDictionary = resourceFactories.getMetricDictionary();
+        LinkedHashSet<Dimension> dimensions = new LinkedHashSet<>();
+        LinkedHashSet<PhysicalTable> physicalTables = new LinkedHashSet<>();
+        LinkedHashSet<ApiMetricName> apiMetricNames = new LinkedHashSet<>();
+        configTable.get("dimensions").forEach(
+                node -> dimensions.add(resourceFactories.getDimension(node.textValue()))
+        );
+        configTable.get("physicalTables").forEach(
+                node -> physicalTables.add(resourceFactories.getPhysicalTable(node.textValue()))
+        );
+        configTable.get("metrics").forEach(
+                node -> {
+                    /* go to the LIP to retrieve the metric */
+                    /* Must be done after having actual metric builders! */
+                    // apiMetricNames.add(new LuthierApiMetricName(node.textValue(), granularity));
+                    // metricDictionary.add();
+                }
+        );
+        TableGroup tableGroup = new TableGroup(
+                physicalTables,
+                apiMetricNames,
+                dimensions
+        );
+
+        configTable.get("granularities").forEach(
+                (JsonNode node) -> {
+                    ZonedTimeGrain granularity = DefaultTimeGrain.valueOf(
+                            node.textValue().toUpperCase(Locale.US)
+                    ).buildZonedTimeGrain(DateTimeZone.forID(configTable.get("dateTimeZone").textValue()));
```

Comment:
The formatting here can be improved abit. First, I would pull `DateTimeZone.forID(configTable.get("dateTimeZone").textValue())`
into its own variable `timeZone`. Then, I would wrap the `granularity` call on the method invocations, so:

```java
DateTimeZone timeZone = DateTimeZone.forID(configTable.get("dateTimeZone").textValue());
ZonedTimeGrain granularity = DefaultTimeGrain
.valueOf(node.textValue().toUpperCase(Locale.US))
.buildZonedTimeGrain(timeZone);
```
I generally find that it can be difficult to parse nested function invocations (i.e. `f(g(h(0, 1), 2))` . Better to
flatten things by pulling out a local variable or two. Furthermore, when I'm invoking two more methods in a single statement,
I often find it clearer to put each method invocation on its own line, rather than trying to wrap the arguments of one
method, but not the other.

Reply:
I agree. Changed in the new commit.